x-pack/plugins/security_solution/public/detections/pages/detection_engine/detection_engine.tsx

```diff
@@ -168,7 +184,31 @@ const DetectionEnginePageComponent: React.FC<DetectionEngineComponentProps> = ({
     sourcererDataView,
     runtimeMappings,
     loading: isLoadingIndexPattern,
+    indexPattern,
   } = useSourcererDataView(SourcererScopeName.detections);
+  const spaceId = useSpaceId();
```

Comment:
Could you please create a separate wrapper which can be called `DetectionPageFilter` which will contain all this new logic. 

It is better to avoid including all this new logic in detection engine page directly as it is already too big. Thanks.

Reply:
Moved this to [detection_engine_filters.tsx](https://github.com/umbopepato/kibana/blob/3c3e8e13fc3b79e7cbf7ebad5a425e1a42581be2/x-pack/plugins/security_solution/public/detections/components/detection_engine_filters/detection_engine_filters.tsx)

Reply:
Thanks for creating a wrapper component.